akka-projection-core/src/main/scala/akka/projection/OffsetManagement.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.projection
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.actor.typed.pubsub.Topic
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.annotation.InternalApi
+import akka.util.Timeout
+
+object OffsetManagement extends ExtensionId[OffsetManagement] {
+  def createExtension(system: ActorSystem[_]): OffsetManagement = new OffsetManagement(system)
+
+  def get(system: ActorSystem[_]): OffsetManagement = apply(system)
+}
+
+class OffsetManagement(system: ActorSystem[_]) extends Extension {
+  private implicit val sys: ActorSystem[_] = system
+  private implicit val askTimeout: Timeout = 10.seconds // FIXME configuration
+  private implicit val ec: ExecutionContext = system.executionContext
+
+  private val topics =
+    new ConcurrentHashMap[String, ActorRef[Topic.Command[ProjectionBehavior.OffsetManagementCommand]]]()
+
+  private def topicName(projectionName: String): String =
+    "projection-" + projectionName
+
+  private def topic(projectionName: String): ActorRef[Topic.Command[ProjectionBehavior.OffsetManagementCommand]] = {
+    topics.computeIfAbsent(projectionName, _ => {
+      val name = topicName(projectionName)
+      system.systemActorOf(Topic[ProjectionBehavior.OffsetManagementCommand](name), name)
+    })
+  }
+
+  /**
+   * INTERNAL API: ProjectionBehavior registers when started.
+   */
+  @InternalApi private[akka] def register(
+      projectionId: ProjectionId,
+      projection: ActorRef[ProjectionBehavior.OffsetManagementCommand]): Unit = {
+    topic(projectionId.name) ! Topic.Subscribe(projection)
+  }
+
+  def getAllProjectionIds(projectionName: String): Future[Set[ProjectionId]] = {
+    // FIXME publish to the topic and gather all replies, or if we know that we are running with ShardedDaemonProcess we know how many
+    ???
+  }
```

Comment:
I wouldn't worry too much about this - from a programmatic management perspective, users will know what projections exist. If/when general purpose management utilities are added (eg, if a remote interface was exposed by akka-management, or a UI was added, etc), then this will be more important, but maybe save the feature till then. For our purposes, we'll probably just wrap the sharded daemon api with our own thing that stores all the registered projections.